ocp_resources/data_science_cluster.py

```diff
@@ -0,0 +1,88 @@
+from ocp_resources.resource import Resource
+
+
+class DataScienceCluster(Resource):
+    """
+    DataScienceCluster Resource for RHOAI operator
+
+    https://access.redhat.com/documentation/en-us/red_hat_openshift_ai_self-managed/2.5/html/installing_and_uninstalling_openshift_ai_self-managed/index
+    """
+
+    api_group = Resource.ApiGroup.DATASCIENCE_CLUSTER
+
+    class ManagementState:
+        """
+        Supported Management States of components of DataScienceCluster.
+
+        Managed: The Operator actively manages the component, installs it, and tries to keep it active.
+        The Operator will upgrade the component only if it is safe to do so.
+
+        Removed: The Operator actively manages the component but does not install it. If the component is
+        already installed, the Operator will try to remove it.
+        """
+
+        MANAGED = "Managed"
+        REMOVED = "Removed"
+
+    def __init__(
+        self,
+        name=None,
```

Comment:
name, client and yaml_file should be removed (should be passed as part of `kwargs`)